test/unit_tests/test_metadata_table.py

```diff
@@ -0,0 +1,141 @@
+#! /usr/bin/env python
+#-----------------------------------------------------------------------
+# Description:  Contains unit tests for parse_metadata_file in the class
+#               MetadataTable in scripts file metadata_table.py
+#
+# Assumptions:  
+#
+# Command line arguments: none
+#
+# Usage: python test_metadata_table.py         # run the unit tests
+#-----------------------------------------------------------------------
+import sys
+import os
+import logging
+import unittest
+
+root_dir = os.path.dirname(os.path.abspath(__file__))
+scripts_dir = os.path.join(root_dir, "../../scripts")
+sample_files_dir = os.path.join(root_dir, "sample_files")
+
+if not os.path.exists(scripts_dir):
+    raise ImportError("Cannot find scripts directory")
+
+sys.path.append(scripts_dir)
+
+from metadata_table import MetadataTable
+
+'''Test parse_metadata_file in metadata_table.py'''
+
+class MetadataTableTestCase(unittest.TestCase):
+
+   """Tests for `parse_metadata_file`."""
+
+   def test_good_host_file(self):
+       """Test that good host file test_host.meta returns one header named test_host"""
+       #Setup
+       known_ddts = list()
+       logger = None
+       filename = os.path.join(sample_files_dir, "test_host.meta")
+       #Exercise
+       result = MetadataTable.parse_metadata_file(filename, known_ddts, logger)
+       #Verify that size of returned list equals number of headers in the test file
+       #       and that header name is 'test_host'
+       self.assertEqual(len(result), 1)
+       listToStr = " ".join([elem.title for elem in result])
+       self.assertIn('test_host', listToStr, msg="Header name 'test_host' is expected but not found")
+
+   def test_good_multi_ccpp_arg_table(self):
+       """Test that good file with 4 ccpp-arg-table returns 4 headers"""
+       known_ddts = list()
+       logger = None
+       filename = os.path.join(sample_files_dir, "test_multi_ccpp_arg_tables.meta")
+
+       result = MetadataTable.parse_metadata_file(filename, known_ddts, logger)
+
+       #Verify that size of returned list equals number of headers in the test file
+       self.assertEqual(len(result), 4)
+       listToStr = " ".join([elem.title for elem in result])
+       #print(listToStr)
```

Comment:
I think this may be susceptible to corruption. For instance, what if one of the headers came back `'make_ddt_run_type'`? That would be wrong but would pass your assert tests below. I suggest just using the title list without converting it to a string, e.g., 
```
titles = [elem.title for elem in result]
```